src/test/java/codeu/controller/LogoutServletTest.java

```diff
@@ -25,29 +25,28 @@
   @Before
   public void setup() {
     logoutServlet = new LogoutServlet();
-    mockSession = Mockito.mock(HttpSession.class);
     mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockSession = Mockito.mock(HttpSession.class);
     mockResponse = Mockito.mock(HttpServletResponse.class);
     mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
-    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/logout.jsp"))
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/login.jsp"))
       .thenReturn(mockRequestDispatcher);
   }
 
   @Test
   public void testDoGetSession() throws IOException, ServletException {
-    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+    Mockito.when(mockRequest.getSession(false)).thenReturn(mockSession);
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("testUser");
     logoutServlet.doGet(mockRequest, mockResponse);
-    // Mockito.verify(mockResponse).sendRedirect("/login");
-    // If there was a session the user should have been redirected to the login page
-    // GIVES ERROR: "Actually, there were zero interactions with this mock"
-    }
+    //Assert.assertEquals("Logout successful!", mockRequest.getAttribute("error"));
```

Comment:
I think this is because you didn't create a Mockito.when() for this method call. I think what you should probably verify instead is the setAttribute() call that is made by the servlet.  I think you should be able to do that in a similar way to how the forward call is verified below.